sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnarBatch.java

```diff
@@ -505,18 +500,12 @@ public void filterNullsInColumn(int ordinal) {
     nullFilteredColumns.add(ordinal);
   }
 
-  private ColumnarBatch(StructType schema, int maxRows, MemoryMode memMode) {
+  public ColumnarBatch(StructType schema, ColumnVector[] columns, int capacity) {
     this.schema = schema;
-    this.capacity = maxRows;
-    this.columns = new ColumnVector[schema.size()];
+    this.columns = columns;
+    this.capacity = capacity;
```

Comment:
Does `capacity` really mean anything in here anymore since the `ColumnVectors` are allocated and populated outside now?  Could we just initialize `this.numRows = 0` and delay initializing of `this.filteredRows` until `setNumRows()` is called?